raster-foundry/raster-foundry
# Fix bulk create users endpoint

## Overview

Fix bulk create users endpoint by upgrading `sttp`.

Before this change, this runtime error was thrown:

java.lang.NoSuchMethodError: io.circe.Json.pretty(Lio/circe/Printer;)Ljava/lang/String;
at sttp.client3.circe.SttpCirceApi.$anonfun$circeBodySerializer$1(SttpCirceApi.scala:16)

### Checklist


### Demo
```
11:35 $ http -v --auth-type=jwt POST :9000/api/users/bulk-create count:=1 peudoUserNameType=HARRY_POTTER organizationId=dfac6307-b5ef-43f7-beda-b9f208bb7726 platformId=31277626-968b-4e40-840b-559d9c67863c grantAccessToParentCampaignOwner:=false grantAccessToChildrenCampaignOwner:=false copyResourceLink:=false
POST /api/users/bulk-create HTTP/1.1
Accept: application/json, */*;q=0.5
Accept-Encoding: gzip, deflate, br
Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6Ik9URXpORGRDUkRRM1JUa3hOVGc1TVVFME9UQTNOekJGTVRZeU0wTkRNekV5TXpoRk1qUXpOQSJ9.eyJodHRwczovL2FwcC5yYXN0ZXJmb3VuZHJ5LmNvbTtwbGF0Zm9ybSI6IjMxMjc3NjI2LTk2OGItNGU0MC04NDBiLTU1OWQ5YzY3ODYzYyIsImh0dHBzOi8vYXBwLnJhc3RlcmZvdW5kcnkuY29tO29yZ2FuaXphdGlvbiI6ImRmYWM2MzA3LWI1ZWYtNDNmNy1iZWRhLWI5ZjIwOGJiNzcyNiIsImh0dHBzOi8vYXBwLnJhc3RlcmZvdW5kcnkuY29tO2Fubm90YXRlQXBwIjp0cnVlLCJodHRwczovL2FwcC5yYXN0ZXJmb3VuZHJ5LmNvbTtzY29wZXMiOiJncm91bmR3b3JrVXNlciIsImh0dHBzOi8vYXBwLnJhc3RlcmZvdW5kcnkuY29tO2dyb3VuZHdvcmtQcm9Vc2VyIjpmYWxzZSwibmlja25hbWUiOiJwY2Fpc3NlK3Rlc3QiLCJuYW1lIjoicGNhaXNzZSt0ZXN0QGF6YXZlYS5jb20iLCJwaWN0dXJlIjoiaHR0cHM6Ly9zLmdyYXZhdGFyLmNvbS9hdmF0YXIvZGE3NWVmNWM0OTdiN2Y1MWRmZGMwMjVlNzI5ODZhNWQ_cz00ODAmcj1wZyZkPWh0dHBzJTNBJTJGJTJGY2RuLmF1dGgwLmNvbSUyRmF2YXRhcnMlMkZwYy5wbmciLCJ1cGRhdGVkX2F0IjoiMjAyMS0xMS0wNFQxNDoxMDozNS42MDJaIiwiZW1haWwiOiJwY2Fpc3NlK3Rlc3RAYXphdmVhLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjpmYWxzZSwiaXNzIjoiaHR0cHM6Ly9yYXN0ZXItZm91bmRyeS1kZXYuYXV0aDAuY29tLyIsInN1YiI6ImF1dGgwfDYxMjhmOTI1MTk4MDM4MDA2OGRmOTNmZCIsImF1ZCI6IkJOQUpyWmMyaFNHN202MERkQlN3aGdoTTlCdlJlVDR1IiwiaWF0IjoxNjM2MDM1MDM4LCJleHAiOjE2MzYwNzEwMzgsIm5vbmNlIjoiVmtkUWRsaFZhVWR2YnpZMGQyeEpZbGwyU25FdWNraE5NbVp6U1haQk4xZHpXR3hLVWxKTmVrSkdNdz09In0.GQkzx_BH7NyPfotaHMNvUJTjvYm0Zey1IQD14o5NLTPTlG2VDDqbsCaPNdeX5FEV7G7C8GgITrEZmPh5X5yerUMwmecmGBR7x2x5n4KDZn1useaHaAqEhIL745QSAcb6e03i8OXRLUyMtetKpbEtnE8py9G64Lzm7dJ-llMcbGXkuEmGQS266tGCPvwC_f9qV1Lj4btgs3Bcqc0wCVoMQLc37yAmx-ObCGKfB5e5oulGK501g6wYg0nkOldb3Cpt-OPCcWEi47QXrdXZS5g1Yd9zrZ6EIvcD5KCJRP6a9JGZQYwPrEvSMIF4LHfr1VuN9R8xy3eK71FUvG792sXD1g
Connection: keep-alive
Content-Length: 276
Content-Type: application/json
Host: localhost:9000
User-Agent: HTTPie/2.1.0

{
    "copyResourceLink": false,
    "count": 1,
    "grantAccessToChildrenCampaignOwner": false,
    "grantAccessToParentCampaignOwner": false,
    "organizationId": "dfac6307-b5ef-43f7-beda-b9f208bb7726",
    "peudoUserNameType": "HARRY_POTTER",
    "platformId": "31277626-968b-4e40-840b-559d9c67863c"
}

HTTP/1.1 200 OK
Content-Length: 26
Content-Type: application/json
Date: Thu, 04 Nov 2021 15:35:43 GMT
Server: akka-http/10.2.6

[
    "amelia-bones-0cc6-442f"
]
```

## Testing Instructions
* Log into Groundwork and get a JWT
* Make sure that user has `users:bulkCreate` in their `scopes`. I ran this SQL command:
```
update users set scopes = '"groundworkUser;users:bulkCreate;campaigns:clone;scen
es:create;projects:addScenes;projects:read;annotationProjects:create;campaigns:create;campaigns:
share;campaigns:readPermissions;projects:update;annotationProjects:createTaskGrid"' where email
= 'pcaisse+test@azavea.com';
```
* Export JWT (`export JWT_AUTH_TOKEN=<jwt>`)
* Make a request to the bulk user creation endpoint like so and it should succeed:
```
http -v --auth-type=jwt POST :9000/api/users/bulk-create count:=1 peudoUserNameType=HARRY_POTTER organizationId=dfac6307-b5ef-43f7-beda-b9f208bb7726 platformId=31277626-968b-4e40-840b-559d9c67863c grantAccessToParentCampaignOwner:=false grantAccessToChildrenCampaignOwner:=false copyResourceLink:=false
```

Closes #5632
